### DISCUSSION TITLE
How do I listen for node clicks on Graphs?

### DISCUSSION THREAD

**Discussion by ndtreviv** [f5/unovis#104] [Q&A] [answered]
Firstly: I really love this library! It's so simple to use. Thank you for creating it!

I want to use some graphs to display data from my graph database. As part of this, I want to be able to click on a node or on a [side label](https://unovis.dev/docs/networks-and-flows/Graph#on-the-side) and then load some accompanying data from another database, or expand the graph another level.

Unfortunately I can't find examples of this in the documentation.
Is it possible?
Thanks for any help offered.

**Comment by rokotyan** [comment, marked as answer]
> Firstly: I really love this library! It's so simple to use.

Thanks @ndtreviv, that's exactly what we've tried to achieve!

> As part of this, I want to be able to click on a node or on a [side label](https://unovis.dev/docs/networks-and-flows/Graph#on-the-side) and then load some accompanying data from another database, or expand the graph another level.

Generally you can achieve that via data update. We've designed the Graph component to smoothly transition between config and data changes, so you should be able to simply pass an updated data array to `Graph` and get it updated.

Check out our [Parallel Layout Graph example](https://unovis.dev/gallery/view?collection=Networks%20and%20Flows&title=Parallel%20Layout%20Graph) where you can click on a node and it'll expand:
![Screen Recording 2023-01-18 at 1 07 33 PM](https://user-images.githubusercontent.com/755708/213295173-bd331b81-bbae-4ec4-a3c3-0a73bbda0753.gif)

There's also a way to control the animation by specifying the coordinates and scale for appearing and disappearing nodes. This part currently undocumented but here are the config options you can play with:
<img width="1194" alt="image" src="https://user-images.githubusercontent.com/755708/213295667-9e118200-b49c-4148-91e4-7463b9f0a98c.png">

**Reply by ndtreviv** [reply]
Yes! Thank you! This was the bit I was missing:
```
          events={{
          [Graph.selectors.node]: {
            click: (d) => {
              console.log("click", d);
            }
          }
        }}
```

Where is this in the documentation? I can't seem to find it...

Thanks again!

**Reply by rokotyan** [reply]
Looks like events are not covered in the docs properly, we'll plan to add a page for it!

**Reply by ndtreviv** [reply]
Is `doubleClick` an event?
Can you please give me a list of the event types supported?

**Reply by ndtreviv** [reply]
For future visitors, the events section is here: https://unovis.dev/docs/networks-and-flows/Graph#events

**Reply by rokotyan** [reply]
@ndtreviv All native [JavaScript Mouse Events](https://developer.mozilla.org/en-US/docs/Web/API/Element#mouse_events) are supported. You should be able to use `dblclick` for a double click.
